core/primitives-core/src/account.rs

```diff
@@ -171,10 +231,15 @@ impl Account {
     }
 
     #[inline]
-    pub fn set_code_hash(&mut self, code_hash: CryptoHash) {
+    pub fn set_local_code_hash(&mut self, code_hash: CryptoHash) {
```

Comment:
same here, replace with `set_contract` accepting `AccountContractType`.
when global contract is set for `Self::V1` you need to upgrade it to `V2` and not to panic